src/rust/ide/view/graph-editor/src/component/node/output/port.rs

```diff
@@ -406,18 +455,37 @@ impl Model {
         shape.set_port_count(port_count);
         shape.set_padding_left(padding_left);
         shape.set_padding_right(padding_right);
+        self.shape = Some(shape.clone());
 
-        self.init_frp(&shape,styles);
-        self.shape = Some(shape);
-        (self.shape.as_ref().unwrap().clone_ref(),self.frp.as_ref().unwrap().clone_ref())
+        let type_label = app.new_view::<text::Area>();
+        let offset_y = styles.get_number(ensogl_theme::graph_editor::node::type_label::offset_y);
+        type_label.set_position_y(offset_y);
+        self.type_label = Some(type_label.clone());
+
+        let display_object = display::object::Instance::new(logger);
+        display_object.add_child(&shape);
+        display_object.add_child(&type_label);
+        self.display_object = Some(display_object.clone());
+
+        self.port_count = max(port_count, 1);
+        self.port_index = port_index;
+
+        self.init_frp(&shape,&type_label,styles);
+        (display_object,self.frp.as_ref().unwrap().clone_ref())
     }
 
-    fn init_frp(&mut self, shape:&PortShapeView, styles:&StyleWatch) {
-        let frp     = Frp::new();
-        let network = &frp.network;
-        let events  = shape.events();
-        let opacity = Animation::<f32>::new(network);
-        let color   = color::Animation::new(network);
+    fn init_frp(&mut self,shape:&PortShapeView,type_label:&text::Area,styles:&StyleWatch) {
+        let frp                = Frp::new();
+        let network            = &frp.network;
+        let events             = shape.events();
+        let opacity            = Animation::<f32>::new(network);
+        let color              = color::Animation::new(network);
+        let type_label_opacity = Animation::<f32>::new(network);
+        let port_count         = self.port_count;
+        let port_index         = self.port_index;
+        let full_type_timer    = DelayedAnimation::new(network);
+        full_type_timer.set_delay(FULL_TYPE_ONSET_DELAY_MS);
+        full_type_timer.set_duration(0.0);
```

Comment:
What are you creating animation for if you are setting its duration to 0 ?

Reply:
I use it just for the delay. When the delay is over, I want to switch the value instantly. I copied this from how the way how visualizations are shown on hover. Is there a better way to do this?